vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java

```diff
@@ -592,46 +594,57 @@ private static void loggingException(Exceptions.RunnableThrowingIOException runn
 
     // ---------------------------------------------Document Operations ----------------------------------------
 
-    @FunctionalInterface
-    interface Operation {
+    private static abstract class Operation {
+
+        private final Lock lock = new ReentrantLock();
+        private final HttpRequest request;
+        private final ResponseHandler handler;
+        private Supplier<Boolean> operation;
+
+        Operation(HttpRequest request, ResponseHandler handler) {
+            this.request = request;
+            this.handler = handler;
+        }
 
         /**
          * Attempts to dispatch this operation to the document API, and returns whether this completed or not.
          * This return {@code} true if dispatch was successful, or if it failed fatally; or {@code false} if
          * dispatch should be retried at a later time.
          */
-        boolean dispatch();
+        boolean dispatch() {
+            if ( ! lock.tryLock())
+                throw new IllegalStateException("Comcurrent attempts at dispatch — this is a bug");
```

Comment:
This shouldn't happen, so throw on it. It would cause multiple operations to be sent to the backend for a single request. Can guard here against buggy usage of this class.